src/libraries/System.Threading.Tasks/tests/Task/TaskRtTests_Core.cs

```diff
@@ -390,6 +390,49 @@ public static void RunTaskCompletionSourceTests_SetException()
             }
         }
 
+        [Fact]
+        public static void RunTaskCompletionSourceTests_SetCanceled()
+        {
+            CancellationTokenSource cts = new CancellationTokenSource();
+            TaskCompletionSource<int> tcs = new TaskCompletionSource<int>();
+
+            tcs.SetCanceled(cts.Token);
+
+            if (tcs.Task.Status != TaskStatus.Canceled)
+            {
+                Assert.True(false, string.Format("RunTaskCompletionSourceTests:    > Error!  Canceled, Status should be Canceled, is {0}", tcs.Task.Status));
+            }
+
+            try
+            {
+                var _ = tcs.Task.Result;
+                Assert.True(false, string.Format("RunTaskCompletionSourceTests:    > Error!  Canceled, but get-Result threw no exception"));
+            }
+            catch(AggregateException exc)
+            {
+                var inner = exc.InnerException as TaskCanceledException;
+                Assert.Equal(inner.CancellationToken, cts.Token);
+            }
+            catch
+            {
+                Assert.True(false, string.Format("RunTaskCompletionSourceTests:    > Error!  Canceled, but get-Result threw unexpected exception"));
+            }
```

Comment:
This can just be:
```C#
AggregateException ae = Assert.Throws<AggregateException>(() => tcs.Task.Result);
TaskCanceledException tce = Assert.IsType<TaskCanceledException>(ae.InnerException);
Assert.Equal(cts.Token, tce.CancellationToken);
```